marklogic-client-api/src/main/java/com/marklogic/client/datamovement/impl/WriteBatcherImpl.java

```diff
@@ -1331,4 +1335,20 @@ public boolean awaitCompletion(long timeout, TimeUnit unit) throws InterruptedEx
       return tasks;
     }
   }
+
+
+@Override
+public WriteBatcher withDefaultMetadata(DocumentMetadataHandle handle) {
+    this.docMetadataHandle = handle;
+    return this;
+}
+
+@Override
+public void addAll(Stream<? extends DocumentWriteOperation> operations, JacksonCSVSplitter splitter) {

```

Comment:
The addAll() method should only take the first parameter.  The stream of operations could be generated by any of many different sources include other kinds of splitters that we add in the future.

I recognize that the goal was to increment the count, but that's better done by the mapping operation within the splitter (so the two classes are loosely coupled).

That also makes it possible to simplify the iteration by calling the forEach() stream method, passing a reference to the add method. That would eliminate the need to construct an iterator.